docs/dev/tools/common-errors.md

```diff
@@ -213,6 +213,19 @@ resource build error: unknown resource type: API "rdk:component:camera" with mod
 1. Check the logs for other errors.
    It may be that the instantiation of the model is failing, for other reasons such as the hardware being disconnected.
 
+### Could not connect to machine part
+
+**Full Error:** `Could not connect to machine part: error updating resources: rpc error: code = ResourceExhausted desc = too many in-flight requests`
```

Comment:
```suggestion
**Full Error:** `Could not connect to machine part: error updating resources: rpc error: code = ResourceExhausted desc = exceeded request limit 100 on resource viam.robot.v1.RobotService`
```

The error message has been updated and is parameterized, so `100` and `viam.robot.v1.RobotService` will change depending on the configured limit and which resource hit the limit.

Also I'm not sure how to fully reproduce the conditions that led to the error in your original text. My suggestion is an educated guess at what the full error will look like made by replacing the last `desc` field with the text from our new error type.